Source/ARTRealtime.m

```diff
@@ -554,10 +554,10 @@ - (void)onConnected:(ARTProtocolMessage *)message {
     // Resuming
     if (_resuming) {
         if (![message.connectionId isEqualToString:self.connection.id]) {
-            [self.logger warn:@"R:%p ARTRealtime: connection has reconnected, but resume failed. Detaching all channels", self];
-            // Fatal error, detach all channels
+            [self.logger warn:@"R:%p ARTRealtime: connection has reconnected, but resume failed. Reattaching any attached channels", self];
+            // Reattach all channels
             for (ARTRealtimeChannel *channel in self.channels) {
-                [channel detachChannel:[ARTStatus state:ARTStateConnectionDisconnected info:message.error]];
+                [channel reattachWithReason:message.error callback:nil];
```

Comment:
It would be cleaner, probably, as it's a connection state change side effect so it's weird the side effect happens before the state change. But since there's no dependency on the state change from the reattach it doesn't matter that much.